public/apps/configuration/panels/tenant-list/tenant-list.test.tsx

```diff
@@ -0,0 +1,272 @@
+  });
+
+  it('delete tenant error', (done) => {
+    mockTenantUtils.requestDeleteTenant.mockImplementationOnce(() => {
+      throw new Error();
+    });
+    // Hide the error message
+    const loggingFunc = jest.fn();
+    jest.spyOn(console, 'log').mockImplementationOnce(loggingFunc);
+    shallow(
+      <TenantList
+        coreStart={mockCoreStart as any}
+        navigation={{} as any}
+        params={{} as any}
+        config={config as any}
+      />
+    );
+    const deleteFunc = useDeleteConfirmState.mock.calls[0][0];
+
+    deleteFunc();
+
+    process.nextTick(() => {
+      expect(loggingFunc).toBeCalled();
+      done();
+    });
+  });
+
+  it('submit tenant', () => {
+    const component = shallow(
+      <TenantList
+        coreStart={mockCoreStart as any}
+        navigation={{} as any}
+        params={{} as any}
+        config={config as any}
+      />
+    );
+    component.find('#createTenant').simulate('click');
+    const submitFunc = component.find(TenantEditModal).prop('handleSave');
+    submitFunc('tenant_1', '');
+
+    expect(mockTenantUtils.updateTenant).toBeCalled();
+  });
+
+  it('submit tenant error', () => {
+    const consoleLog = jest.spyOn(console, 'log');
+    consoleLog.mockImplementation(() => {});
+    const error = new Error();
+    mockTenantUtils.updateTenant.mockImplementationOnce(() => {
+      throw error;
+    });
+    const component = shallow(
+      <TenantList
+        coreStart={mockCoreStart as any}
+        navigation={{} as any}
+        params={{} as any}
+        config={config as any}
+      />
+    );
+    component.find('#createTenant').simulate('click');
+    const submitFunc = component.find(TenantEditModal).prop('handleSave');
+    submitFunc('tenant_1', '');
+
+    // Expect error log
+    expect(consoleLog).toBeCalledWith(error);
+  });
+
+  describe('Action menu enable-diable check', () => {
+    const sampleTenantList: Tenant[] = [
+      {
+        tenant: 'tenant_1',
+        description: '',
+        reserved: true,
+        tenantValue: 'tenant_1',
+      },
+      {
+        tenant: 'tenant_2',
+        description: '',
+        reserved: false,
+        tenantValue: 'tenant_2',
+      },
+    ];
+    it('edit and delete should be disabled when there is at least 1 reserved tenant selected', () => {
+      jest.spyOn(React, 'useState').mockImplementation(() => [[sampleTenantList], jest.fn()]);
+      const component = shallow(
+        <TenantList
+          coreStart={mockCoreStart as any}
+          navigation={{} as any}
+          params={{} as any}
+          config={config as any}
+        />
+      );
+      expect(component.find('#edit').prop('disabled')).toBe(true);
+      expect(component.find('#delete').prop('disabled')).toBe(true);
+    });
+
+    it('All menues should be disabled when there is multiple tenant selected including multiple tenant', () => {
+      jest.spyOn(React, 'useState').mockImplementation(() => [[sampleTenantList], jest.fn()]);
```

Comment:
Minor: since this sampleTenantList is same as the test above, it may not be easy to tell whether the expected disabled status is caused by reserve or multiple selection. Probably better to have sampleTenantList with two non reserve tenants.

Reply:
Agreed. Added UT for sampleTenantList with two non reserve tenants.